app/services/cap_quantity.rb

```diff
@@ -42,7 +42,7 @@ def reload_order
   end
 
   def unavailable_stock_lines_for
-    order.line_items.where('variant_id NOT IN (?)', available_variants_for.select(&:id))
+    order.line_items.where.not(variant_id: available_variants_for.select(&:id))
```

Comment:
This breaks a spec but I haven't found out yet why.

Reply:
Here's the difference in generated SQL:
```diff
-(variant_id NOT IN (NULL))
+1=1
```

The result of `available_variants_for.select(&:id)` is an empty array `[]`. So I guess that the difference is that that `1=1` is always true while `variant_id NOT IN (NULL)` is always false. I don't know which one is correct though.

I think, when no variants are available then all variants should be in `unavailable_stock_lines_for`. So your new version is actually fixing a bug. But that case needs to be added to the spec for this service and the currently failing spec needs to be fixed by making the line items available for shipping.